### ISSUE TITLE
Fails with HTTP 500 when repo is not public

### ISSUE THREAD

**Issue by Leonidas-from-XIV** [ocurrent/ocurrent#422]
I added a non-public repo to the CI Github app (which github happily let's you do) and it [shows up in the UI](https://ocaml.ci.dev/github/Leonidas-from-XIV/mucppo/commit/cab248f3e5ba2f30e3c246e4dcd2bf64998de085). Unfortunately, all builds fail and when attempting to look at the [details](https://ocaml.ci.dev/github/Leonidas-from-XIV/mucppo/commit/cab248f3e5ba2f30e3c246e4dcd2bf64998de085/variant/%28analysis%29) the page fails with an HTTP 500 page that doesn't tell me anything on why it failed.

@kit-ty-kate confirmed my suspicion that this is due to the repo being non-public: https://ocaml.ci.dev:8100/job/2023-04-27/130807-git-fetch-a5ebdd

I don't know what the policy about handling non-public repos is, but it would be nice to at least have an error message that says it is failing because the repo is not public and to have it work I'd need to switch it to be public.

**Comment by maiste** [comment]
Indeed, we do not support private repositories. There is an attempt to handle them in `ocurrent` but in fortunately we detect the issue when we try to build them actually, not before.

**Comment by tmcgilchrist** [comment]
It should be possible to retrieve extra data about GitHub repositories and either not add them to be built or show an appropriate error message. 

A combination of not registering the repo here https://github.com/ocurrent/ocaml-ci/blob/master/service/pipeline.ml#L73-L81 and ignoring the web hooks at https://github.com/ocurrent/ocurrent/blob/acaa4181fdb9fbcb4db9052bd5e8f57f60de3ec7/plugins/github/current_github.ml#L54-L66 could work.
